tests/integrations/raftstore/test_merge.rs

```diff
@@ -1450,3 +1450,36 @@ fn test_merge_pessimistic_locks_repeated_merge() {
         .unwrap();
     assert_eq!(value, lock.into_lock().to_bytes());
 }
+
+#[test]
+fn test_stale_message_after_merge() {
+    let mut cluster = new_server_cluster(0, 3);
+    configure_for_merge(&mut cluster);
+    cluster.run();
+    let pd_client = Arc::clone(&cluster.pd_client);
+    pd_client.disable_default_operator();
+
+    cluster.must_transfer_leader(1, new_peer(1, 1));
+
+    cluster.must_put(b"k1", b"v1");
+    cluster.must_put(b"k3", b"v3");
+
+    let region = cluster.get_region(b"k1");
+    cluster.must_split(&region, b"k2");
+    let left = cluster.get_region(b"k1");
+    let right = cluster.get_region(b"k3");
+
+    pd_client.must_remove_peer(left.get_id(), find_peer(&left, 3).unwrap().to_owned());
```

Comment:
Why not use four stores and isolate the deleted node?

Reply:
We need to simulate the case that the peer id from the same store is increased.

Reply:
can we reproduce the case to generate the message instead of building a dummy message?

Reply:
We can't, see https://github.com/tikv/tikv/issues/12023#issuecomment-1055143820

Reply:
> We need to simulate the case that the peer id from the same store is increased.

For example, (1, 2, 3, 4) -> (1, 2, 3) -> (1, 2, 5). And 4 is isolated until merged is finished.

Reply:
Then peer 4's message won't trigger this panic because `util::find_peer(region, from_store_id)` will return `None`